src/js/yAxis.js

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    // Mix-in grade for value-based y axis
+    // should be used alongside the floe.chartAuthoring.axis
+    // grade in charts that require a value-based y axis
+
+    fluid.defaults("floe.chartAuthoring.yAxis", {
```

Comment:
At the moment, the barrier is that `floe.chartAuthoring.xAxis` and `floe.chartAuthoring.yAxis` are mix-in grades used in `floe.chartAuthoring.lineChart.timeSeriesMultiDataSet` rather than separate component grades (a legacy on when there was one big `floe.chartAuthoring.lineChart` component before separate drawn elements were split out into mix-ins). So implementing a base grade with `drawAxis()` and `getAxisTickFormat()` invokers in implementing grades would result in one overwriting another.

The `gradeNames` block from `floe.chartAuthoring.lineChart.timeSeriesMultiDataSet` probably makes the issue clearer:

```
gradeNames: ["floe.chartAuthoring.axis", "floe.chartAuthoring.xAxisTimeSeries", "floe.chartAuthoring.yAxis",  "floe.chartAuthoring.lineChart.timeSeries.line",  "floe.chartAuthoring.valueBinding", "floe.svgDrawingArea"]
```

At the moment all of the chart elements need to be part of a single viewComponent (they all share the same SVG drawing area); they're split out into separate mix-in grades to isolate the various drawing functions, if that makes sense.

That said, the component trees are indeed substantially similar; maybe longer-term we want a component-based approach where the drawing area is a view component, and drawn chart elements (axes, etc) are other (non-view) components that contain a reference to the drawing area. I can see this making the charts more composable from various grades without the namespace issues presented by the current mixin approach. If that makes sense I can make a separate JIRA for this issue?